### PR TITLE
docs: make usage of Schema Object consistent

### PR DESCRIPTION
**Pull request by char0n**
**Description**

[Main specification document](https://github.com/asyncapi/spec/blob/master/spec/asyncapi.md) says this:

image

All the Schema Objects in main specification document has following type notation:

image

All binding objects specification documents contain following type notation:

image

---

From this, it is apparent that type notation for fields using `Schema Object`, is not consistent within the spec and probably needs to be clarified. There is no reason (at least none that I can see currently) why to constrain binding objects fields to using `Schema Object` and not allow `Reference Object` in it's place as well. The AsynAPI specification tells us following: `Alternatively, any time a Schema Object can be used, a Reference Object can be used in its place.` - this can also be interpreted that using `Reference Object` in binding objects fields is allowed, but if we check the [schemas](https://github.com/asyncapi/bindings/blob/master/http/json_schemas/message.json#L15), they tell us `Reference Object` is not allowed. As we already [stipulated before](https://github.com/asyncapi/spec/pull/781), the specification is the source of truth and not the schemas.

This PR open possibility to use `Reference Object` in binding objects fields and updates schemas to use latest versions of AsyncAPI  schemas (2.4.0). This also enhanced re-usability aspect of the spec. 

**Related issue(s)**

https://github.com/swagger-api/apidom/issues/1832

### CODE REVIEW

File: kafka/README.md [hunk 1]
```diff
@@ -69,7 +69,7 @@ This object contains information about the message representation in Kafka.
 
 Field Name | Type | Description
 ---|:---:|---
-<a name="messageBindingObjectKey"></a>`key` | [Schema Object][schemaObject] \| [AVRO Schema Object](https://avro.apache.org/docs/current/spec.html) | The message key. **NOTE**: You can also use the [reference object](https://asyncapi.io/docs/specifications/v2.1.0#referenceObject) way. 
+<a name="messageBindingObjectKey"></a>`key` | [Schema Object][schemaObject] \| [Reference Object](referenceObject) \| [AVRO Schema Object](https://avro.apache.org/docs/current/spec.html) | The message key. **NOTE**: You can also use the [reference object](https://asyncapi.io/docs/specifications/v2.1.0#referenceObject) way.
```
Comment (fmvilas) [inline kafka/README.md:72]: I think we can remove this now then:

```suggestion
<a name="messageBindingObjectKey"></a>`key` | [Schema Object][schemaObject] \| [Reference Object](referenceObject) \| [AVRO Schema Object](https://avro.apache.org/docs/current/spec.html) | The message key.
```

**Review by fmvilas — CHANGES_REQUESTED** [review]

File: kafka/README.md (hunk 1 above)
Comment (char0n) [reply kafka/README.md]: Suggestion merged. Thanks!

**Review by fmvilas — DISMISSED** [review]
LGTM 👍

**Review by dalelane — APPROVED** [review]

**Comment by char0n** [comment]
@lbroudoux @GeraldLoeffler @derberg @KhudaDad414 would you be able to look at this PR please? THanks!

**Review by derberg — DISMISSED** [review]

**Review by smoya — DISMISSED** [review]

**Comment by char0n** [comment]
@KhudaDad414 @lbroudoux @GeraldLoeffler pinging again on this one.

**Review by lbroudoux — DISMISSED** [review]
Looks good to me and makes a lot of sense

**Review by KhudaDad414 — APPROVED** [review]
LGTM!

**Review by GeraldLoeffler — APPROVED** [review]
excellent - thank you!

**Comment by char0n** [comment]
merged the changes from master, so unfortunately new round of reviews is required.

**Review by KhudaDad414 — APPROVED** [review]
LGTM!

**Review by fmvilas — DISMISSED** [review]

**Review by dalelane — DISMISSED** [review]

**Review by derberg — APPROVED** [review]

**Comment by jonaslagoni** [comment]
Any reason this is not merged? 😆

**Comment by char0n** [comment]
@jonaslagoni I guess only because of my inability to resolve the conflict in this PR for last 2 months ;]

**Review by KhudaDad414 — APPROVED** [review]

**Review by fmvilas — APPROVED** [review]
@derberg @dalelane please approve again 🙏

**Review by dalelane — APPROVED** [review]

**Review by derberg — APPROVED** [review]

**Comment by derberg** [comment]
as multiple  bindings were modified we need below folks to approve again:
- @smoya 
- @lbroudoux 
- @GeraldLoeffler

**Comment by char0n** [comment]
Haha this PR proved to be pretty difficult to merge ;]

**Comment by derberg** [comment]
@smoya
@lbroudoux
@GeraldLoeffler

can you folks have a look 🙏🏼

**Comment by char0n** [comment]
3 more needed ;]

**Review by GeraldLoeffler — APPROVED** [review]

**Review by lbroudoux — APPROVED** [review]
LGTM. Thanks!

**Comment by char0n** [comment]
Thank you everybody for your patience. Finally merged!

**Comment by Tenischev** [comment]
@char0n @derberg @fmvilas 
this looks strange
image

**Comment by smoya** [comment]
> @char0n @derberg @fmvilas this looks strange image

Indeed got merged with the Git merge HEAD marks left. See https://github.com/asyncapi/bindings/pull/143/files#diff-ba2c28c470746625d05c982e58e3b6fdd081ddae606d2918856189bb787c8b4fR148

@char0n are you ok fixing this? Otherwise I'm happy to do it so we do not leave that broken for so much time.

**Comment by derberg** [comment]
isn't it the same as https://github.com/asyncapi/bindings/pull/219 ?

**Comment by char0n** [comment]
@derberg yes, #219 and #220 are the same. Author of #219 closed it as he probably saw #220.

**Pull request merged**